include/ygm/container/oversized_bag.hpp

```diff
@@ -89,19 +309,25 @@ class oversized_bag : public detail::base_async_insert_value<oversized_bag<Item>
   /**
    * @todo we'll need to close all of the files we opened
    */
-  ~oversized_bag() { m_comm.barrier(); }
+  ~oversized_bag() { 
+    m_comm.barrier(); 
+    for (auto &file : m_files) {
+      file.m_file_io.close();
+    }
+  }
 
   /** @todo ----------------------------------------------unchanged---------------------------------------------- */
   oversized_bag(const self_type &other)  // If I remove const it compiles
-      : m_comm(other.comm()), pthis(this), partitioner(other.comm()) {
+      : m_comm(other.comm()), pthis(this), partitioner(other.comm()), m_file_info{other.m_file_info} {
     pthis.check(m_comm);
   }
 
+  /** @todo */
   oversized_bag(self_type &&other) noexcept
```

Comment:
Thats either the move or copy constructor. I think there's a chance we can just not allow it but i implemented them like the regular bag. since the data structure is file backed there's not much use for it.